### ISSUE TITLE
multiple skipIfMatch regexes not working?

### ISSUE THREAD

**Issue by iilei** [aotaduy/eslint-plugin-spellcheck#29]
It seems passing multiple Regexes to skipIfMatch doesn't apply.

This is the related config (yaml) I have:

```yaml
    skipIfMatch:
    - ^\S[^aeiou]{1,3}$    # Intl, cldr
    - https?://[^s]*
```
more config, just in case:
````
  extends: "airbnb"
  root: true
  ecmaFeatures:
    jsx: true
    templateStrings: true
    classes: true
    modules: true
````

and something among the lines of

```js
const cldrDataByLang = [...cldrDataByLocale].reduce((map, [locale, data]) => { 
    /* ... omitted ... */
});
```
Results in
```bash
    3:29  error  You have a misspelled word: cldr on String      spellcheck/spell-checker
```

As 
```js
/^\S[^aeiou]{1,3}$/.test('cldr') === true;
```
I would expect this not to be considered a rule violation.

**Comment by DavidRieman** [comment]
I just found this excellent plugin today.  Having run into some curious behavior myself, I believe our issues are related.  Ultimately, skipIfMatch seems to _match against each entire line_, and skips spell check _for the entire line_, if the match resolves true.  (Personally I believe that skipping by word would generally be preferred, and am about to open another issue on the topic.)

Since skipIfMatch is only matching entire lines, if your line really looks like
`    \\ cldr`
or
`var cldr = require('cldr');`
or even simply having a semicolon or whitespace on either end, then your regular expression will _not_ match that line.  Thus spell check will not be skipped for that entire line.

I think adding the cases like 'cldr' and 'intl' to the skipWords should work better.

**Comment by iilei** [comment]
I am aware of the fact the regex applies to much more and I could easily *fix* my use case by explicitly adding 'cldr' and 'intl' to the skipWords section. Actually I intend to be very lax on Strings that match a pattern often seen in Abbreviations / Acronyms -- loosely determined by the regex above. 

Nevertheless, I used this to demonstrate an unexpected behaviour.

**Comment by aotaduy** [comment]
Hello @iilei,
The issue here is that skipIfMatch will apply to the entire identifier, string, templateString or Comment not to individual words obtained for later processing, in your case skipIfMatch is being applied to the string 'cldrDataByLang' not to individual parts of it such as 'cldr', 'Data', etc. I'm thinking on adding a skipIfMatchWord parameter for this special cases as suggested by david. 

I'll document this on the Readme to avoid confusions with skipWords and skipIfMatch and I will could add the skipIfMatchWord parameter later when I find some time. 
 
Thanks!

**Comment by pubkey** [comment]
Would be much better if I could tell the plugin to ignore 3-letter-words. :(

**Comment by aotaduy** [comment]
Hey @pubkey 
If you have the time to add the parameter to the configuration I will gladly accept a PR for that.

**Comment by aotaduy** [comment]
Closing since pull request merged with new feature
